src/snowflake/cli/api/sql_execution.py

```diff
@@ -137,3 +143,27 @@ def to_fully_qualified_name(self, name: str):
 
         schema = self._conn.schema or "public"
         return f"{self._conn.database}.{schema}.{name}".upper()
+
+    def show_specific_object(
+        self,
+        object_type_plural: str,
+        unqualified_name: str,
+        name_col: str = "name",
+        in_clause: str = "",
+    ) -> Optional[dict]:
+        """
+        Executes a "show <objects> like" query for a particular entity with a
+        given (unqualified) name. This command is useful when the corresponding
+        "describe <object>" query does not provide the information you seek.
+        """
+        show_obj_query = f"show {object_type_plural} like {identifier_to_show_like_pattern(unqualified_name)} {in_clause}".strip()
+        show_obj_cursor = self._execute_query(  # type: ignore
+            show_obj_query, cursor_class=DictCursor
+        )
+        if show_obj_cursor.rowcount is None:
+            raise SnowflakeSQLExecutionError(show_obj_query)
+        show_obj_row = find_first_row(
+            show_obj_cursor,
+            lambda row: row[name_col] == unquote_identifier(unqualified_name),
+        )
+        return show_obj_row
```

Comment:
To me, that's a different layer. My mental model of layers:

- command "spec" / app layer (i.e. `commands.py`)
- command implementation layer (i.e. managers; e.g. `ObjectManager.py`)
- utility layer (e.g. `utils/*.py`, and `SqlExecutionMixin`)

Suppose we were to put it in `ObjectManager`. Other managers that want to use it would have to instantiate an `ObjectManager()`, then call this. Right now there's no args for the constructor but I'm not sure we can count on that going forward given the design (e.g. if we ever needed to support multiple concurrent connections). So this felt like the cleanest place to put it.